src/Core/Logger.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * @file src/Core/Logger.php
+ */
+namespace Friendica\Core;
+
+use Friendica\BaseObject;
+use Friendica\Core\Config;
+use Friendica\Util\DateTimeFormat;
+use ReflectionClass;
+
+/**
+ * @brief Logger functions
+ */
+class Logger extends BaseObject
+{
+    // Log levels:
+    const WARNING = 0;
+    const INFO = 1;
+    const TRACE = 2;
+    const DEBUG = 3;
+    const DATA = 4;
+    const ALL = 5;
+
+    public static $levels = [];
```

Comment:
This array is actually an associative array with the constant value as the key, and the level name as value. You can write it like this:
```php
class Logger 
{
    // Log levels:
    const WARNING = 0;
    const INFO = 1;
    const TRACE = 2;
    const DEBUG = 3;
    const DATA = 4;
    const ALL = 5;
    
     public static $levels = [
         self::WARNING => 'Warning',
         self::INFO => 'Info',
         self::TRACE => 'Trace',
         self::DEBUG => 'Debug',
         self::DATA => 'Data',
         self::ALL => 'All',
     ];
...
```

and get rid of the `getConstants` method.